controllers/autoscaling_test.go

```diff
@@ -0,0 +1,136 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	github2 "github.com/google/go-github/v31/github"
+	"github.com/summerwind/actions-runner-controller/api/v1alpha1"
+	"github.com/summerwind/actions-runner-controller/github"
+	"k8s.io/klog/klogr"
+	"testing"
+)
+
+func TestDetermineDesiredReplicas_RepositoryRunner(t *testing.T) {
+	intPtr := func(v int) *int {
+		return &v
+	}
+
+	testcases := []struct {
+		fixed               *int
+		max                 *int
+		min                 *int
+		workflowRunStatuses []string
+		want                int
+	}{
+		// 3 demanded, max at 3
+		{
+			min:                 intPtr(2),
+			max:                 intPtr(3),
+			workflowRunStatuses: []string{"queued", "in_progress", "in_progress", "completed"},
+			want:                3,
+		},
+		// 3 demanded, max at 2
+		{
+			min:                 intPtr(2),
+			max:                 intPtr(2),
+			workflowRunStatuses: []string{"queued", "in_progress", "in_progress", "completed"},
+			want:                2,
+		},
+		// 2 demanded, min at 2
+		{
+			min:                 intPtr(2),
+			max:                 intPtr(3),
+			workflowRunStatuses: []string{"queued", "in_progress", "completed"},
+			want:                2,
+		},
+		// 1 demanded, min at 2
+		{
+			min:                 intPtr(2),
+			max:                 intPtr(3),
+			workflowRunStatuses: []string{"queued", "completed"},
+			want:                2,
+		},
+		// 1 demanded, min at 2
+		{
+			min:                 intPtr(2),
+			max:                 intPtr(3),
+			workflowRunStatuses: []string{"in_progress", "completed"},
+			want:                2,
+		},
+		// 1 demanded, min at 1
+		{
+			min:                 intPtr(1),
+			max:                 intPtr(3),
+			workflowRunStatuses: []string{"queued", "completed"},
+			want:                1,
+		},
+		// 1 demanded, min at 1
+		{
+			min:                 intPtr(1),
+			max:                 intPtr(3),
+			workflowRunStatuses: []string{"in_progress", "completed"},
+			want:                1,
+		},
+		// fixed at 3
+		{
+			fixed: intPtr(3),
+			want:  3,
+		},
+	}
+
+	for i := range testcases {
+		tc := testcases[i]
+
+		log := klogr.New()
```

Comment:
It may be better to use the controller-runtime log package, as this will add a new dependency package.
https://godoc.org/sigs.k8s.io/controller-runtime/pkg/log.

Reply:
Migrated to the controller-runtime one in 7555b7e. Thanks!